### PR TITLE
[5.1] fix wrong parameter value of new trailingslash parameter in SEF plugin

### PR DESCRIPTION
**Pull request by SniperSister**
### Summary of Changes
The new trailing slash parameter, introduced in #42702 had received another patch that changed the parameter values in the XML, see #43195. That second change was incomplete, leaving the onAfterInitialise function unpatched.

This PR fixes the issue.

### Testing Instructions
See #42702 and/or code review (tbh it's a no-brainer as we are obviously comparing with a non-existing value...)

### Actual result BEFORE applying this Pull Request
* Build rules not attached

### Expected result AFTER applying this Pull Request
* Build rules attached

### CODE REVIEW

**Comment by brianteeman** [comment]
Perfect example of why I said

> Without a test suite that will measure all the types of urls that can be exist now and that this, and any other future changes, can be measured against this should be reverted. Too risky especially as the "gains" are so minor and can be addressed without touching any code as already discussed.

**Comment by SniperSister** [comment]
@brianteeman we are planing a dedicated “router test suite” sprint for that reason :) so your feedback has been heard and appreciated!

**Comment by brianteeman** [comment]
great to here I was partialy listened to. Shame that it has taken precited issues taking place AFTER the merges. Would save a lot of hurt if these _untested_ changes had not been merged until after the tests were created. After all at least one of them was addressing a multiple year old issue

File: plugins/system/sef/src/Extension/Sef.php [hunk 1]
```diff
@@ -70,10 +70,10 @@ public function onAfterInitialise()
             && !$app->get('sef_suffix')
             && $this->params->get('trailingslash')
```
Comment (richard67) [inline plugins/system/sef/src/Extension/Sef.php]: Hmm, won't this line resolve to `false` if the parameter value is zero? It seems this here needs to be changed to `&& $this->params->get('trailingslash', '-1') != '-1'` like we have it elsewhere when checking if the value is not `PLG_SEF_TRAILINGSLASH_OPTION_NONE`. Or maybe better `&& (int)$this->params->get('trailingslash', -1) !== -1`. It seems we have a weird mix of strict and not strict comparisons with the `trailingslash` parameter, so maybe that could be cleaned up.

Comment (richard67) [reply plugins/system/sef/src/Extension/Sef.php]: Or maybe we just can remove that line as the parameter values are checked again inside that if.

File: plugins/system/sef/src/Extension/Sef.php [hunk 2]
```diff
@@ -68,7 +68,7 @@ public function onAfterInitialise()
         if (
             $app->get('sef')
             && !$app->get('sef_suffix')
-            && $this->params->get('trailingslash')
+            && $this->params->get('trailingslash', -1) !== -1
```
Comment (richard67) [inline plugins/system/sef/src/Extension/Sef.php]: Hmm, elsewhere we either use non-strict comparison, or we cast it to int when using strict comparison.
```suggestion
            && $this->params->get('trailingslash', -1) != -1
```

**Comment by richard67** [comment]
I have tested this item :white_check_mark: successfully on 81b7afeda08c2dce8aec9b34267220d086e6af0b<br /><br />I've tested as follows:

I've added some error logging to the code to see when which rule is attached and made sure that PHP errors are logged into a log file.

E.g. here the code with the PR applied and with my additional error logs:

```
        if (
            $app->get('sef')
            && !$app->get('sef_suffix')
            && $this->params->get('trailingslash', -1) != -1
        ) {
            if ($this->params->get('trailingslash') == 0) {
                // Remove trailingslash
                $router->attachBuildRule([$this, 'removeTrailingSlash'], SiteRouter::PROCESS_AFTER);
                // DEBUG
                error_log('DEBUG: Rule "removeTrailingSlash" attached.');
            } elseif ($this->params->get('trailingslash') == 1) {
                // Add trailingslash
                $router->attachBuildRule([$this, 'addTrailingSlash'], SiteRouter::PROCESS_AFTER);
                 // DEBUG
                error_log('DEBUG: Rule "addTrailingSlash" attached.');
           }
        }
```

Then I have tested the 3 values for the "trailingslash" parameter of the SEF plugin by selecting that value, sabing the plugin settings and then calling the frontpage and watching my error log.

Without the changes from this PR, only the `DEBUG: Rule "removeTrailingSlash" attached.` can be found in the PHP error log when the value of the "trailingslash" parameter is 1, i.e. "Enforce URLs with trailing slash", or when it is -1, i.e. "No change", but not when it should be, and the `DEBUG: Rule "addTrailingSlash" attached.` can never be seen in the PHP error log.

With the changes from this PR applied, the debug logs appear as expected:
- Nothing when the value of the "trailingslash" parameter is -1, i.e. "No change"
- `DEBUG: Rule "removeTrailingSlash" attached.` when the value of the "trailingslash" parameter is 0, i.e. "Enforce URLs without trailing slash"
- `DEBUG: Rule "addTrailingSlash" attached.` when the value of the "trailingslash" parameter is 1, i.e. "Enforce URLs with trailing slash"
<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/43292">issues.joomla.org/tracker/joomla-cms/43292</a>.</sub>

**Comment by rfmjoe** [comment]
tested this (https://raw.githubusercontent.com/joomla/joomla-cms/81b7afeda08c2dce8aec9b34267220d086e6af0b/plugins/system/sef/src/Extension/Sef.php) successfully on a live-site.

**Comment by richard67** [comment]
@rfmjoe Could you mark your test result in the issue tracker so it's properly counted? For doing this, go to this PR in the issue tracker here https://issues.joomla.org/tracker/joomla-cms/43292 , use the blue "Test this" button at the top left corner, select your test result and submit. Thanks in advance.

**Comment by LadySolveig** [comment]
Thank you @SniperSister 👍🏼  and also for testing @richard67 and @rfmjoe !

**Pull request merged**